src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -184,187 +184,248 @@ public async Task ReinstateSubscriptionAsync(Guid organizationId)
-        {
-            var scimConnections = await _organizationConnectionRepository.GetByOrganizationIdTypeAsync(organization.Id,
-                OrganizationConnectionType.Scim);
-            if (scimConnections != null && scimConnections.Any(c => c.GetConfig<ScimConfig>()?.Enabled == true))
+            if (!newPlan.HasResetPassword && organization.UseResetPassword)
             {
-                throw new BadRequestException("Your new plan does not allow the SCIM feature. " +
-                    "Disable your SCIM configuration.");
+                var resetPasswordPolicy =
+                    await _policyRepository.GetByOrganizationIdTypeAsync(organization.Id, PolicyType.ResetPassword);
+                if (resetPasswordPolicy != null && resetPasswordPolicy.Enabled)
+                {
+                    throw new BadRequestException("Your new plan does not allow the Password Reset feature. " +
+                                                  "Disable your Password Reset policy.");
+                }
             }
-        }
 
-        if (!newPlan.HasCustomPermissions && organization.UseCustomPermissions)
-        {
-            var organizationCustomUsers =
-                await _organizationUserRepository.GetManyByOrganizationAsync(organization.Id,
-                    OrganizationUserType.Custom);
-            if (organizationCustomUsers.Any())
+            if (!newPlan.HasScim && organization.UseScim)
             {
-                throw new BadRequestException("Your new plan does not allow the Custom Permissions feature. " +
-                                              "Disable your Custom Permissions configuration.");
+                var scimConnections = await _organizationConnectionRepository.GetByOrganizationIdTypeAsync(
+                    organization.Id,
+                    OrganizationConnectionType.Scim);
+                if (scimConnections != null && scimConnections.Any(c => c.GetConfig<ScimConfig>()?.Enabled == true))
+                {
+                    throw new BadRequestException("Your new plan does not allow the SCIM feature. " +
+                                                  "Disable your SCIM configuration.");
+                }
             }
-        }
 
-        // TODO: Check storage?
+            if (!newPlan.HasCustomPermissions && organization.UseCustomPermissions)
+            {
+                var organizationCustomUsers =
+                    await _organizationUserRepository.GetManyByOrganizationAsync(organization.Id,
+                        OrganizationUserType.Custom);
+                if (organizationCustomUsers.Any())
+                {
+                    throw new BadRequestException("Your new plan does not allow the Custom Permissions feature. " +
+                                                  "Disable your Custom Permissions configuration.");
+                }
+            }
 
-        string paymentIntentClientSecret = null;
-        var success = true;
```

Comment:
Leave these where they are, rather than declaring them at the top of the method. I think it's better to have them close to the code that uses them.

Reply:
resolved